Panic happens when TiDB starts with 'skip-grant-table'

## Bug Report

### 1. Minimal reproduce step (Required)

Start a TiDB server with ['skip-grant-table' mode enabled](https://docs.pingcap.com/tidb/stable/user-account-management#forget-the-root-password):

```
(only the root user of the operating system can start the TiDB process with this mode)
sudo $RUN_TIDB
...
```
Try to connect to TiDB:
```
$ mysql -u ... 
```

### 2. What did you expect to see? (Required)
Everything is OK.

### 3. What did you see instead (Required)
A panic is noticed:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x261b08a]

goroutine 757 [running]:
github.com/pingcap/tidb/privilege/privileges.(*Handle).Get(...)
	/home/bb7133/Projects/gopath/src/github.com/pingcap/tidb/privilege/privileges/cache.go:1485
github.com/pingcap/tidb/privilege/privileges.(*UserPrivileges).GetAuthPlugin(0xc010b16900, 0xc010b3c004, 0x4, 0x0, 0x0, 0x1, 0x3855880, 0x4065c00000000001, 0xc010b2e0a0)
	/home/bb7133/Projects/gopath/src/github.com/pingcap/tidb/privilege/privileges/privileges.go:224 +0x2a
github.com/pingcap/tidb/session.(*session).AuthPluginForUser(0xc010337000, 0xc010b2e0a0, 0x0, 0x200, 0xae, 0xc0006b7380)
	/home/bb7133/Projects/gopath/src/github.com/pingcap/tidb/session/session.go:2302 +0xb8
github.com/pingcap/tidb/server.(*clientConn).checkAuthPlugin(0xc010b22000, 0x3fa0270, 0xc010b16090, 0xc0006b7878, 0xae, 0xae, 0x3c, 0x0, 0x0)
	/home/bb7133/Projects/gopath/src/github.com/pingcap/tidb/server/conn.go:810 +0xab
github.com/pingcap/tidb/server.(*clientConn).readOptionalSSLRequestAndHandshakeResponse(0xc010b22000, 0x3fa0270, 0xc010b16090, 0x0, 0x0)
	/home/bb7133/Projects/gopath/src/github.com/pingcap/tidb/server/conn.go:682 +0x289
github.com/pingcap/tidb/server.(*clientConn).handshake(0xc010b22000, 0x3fa0270, 0xc010b16090, 0x3fa0270, 0xc010b16090)
	/home/bb7133/Projects/gopath/src/github.com/pingcap/tidb/server/conn.go:254 +0x41a
github.com/pingcap/tidb/server.(*Server).onConn(0xc0009e3ee0, 0xc010b22000)
	/home/bb7133/Projects/gopath/src/github.com/pingcap/tidb/server/server.go:453 +0xa9
created by github.com/pingcap/tidb/server.(*Server).startNetworkListener
	/home/bb7133/Projects/gopath/src/github.com/pingcap/tidb/server/server.go:404 +0x8fc
```

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->
```
Release Version: v5.2.0-alpha-582-g3d88be834
Edition: Community
Git Commit Hash: 3d88be8344c3dbc6a344efbc69bee3d2297339cb
Git Branch: release-5.2
UTC Build Time: 2021-08-
```

### 5. Comments
When `skip-grant-table` is enabled, the load of privilege is skipped:

https://github.com/pingcap/tidb/blob/4be68210b1435e97c6a00628d1fa09a0ca944a2f/session/session.go#L2600-L2609

So `p.Handle` is `nil`, and TiDB will panic here:

https://github.com/pingcap/tidb/blob/723e2bc6d8513ea8c466f3857c11af34ea71502f/privilege/privileges/privileges.go#L224

A simple check like [this](https://github.com/pingcap/tidb/blob/723e2bc6d8513ea8c466f3857c11af34ea71502f/privilege/privileges/privileges.go#L95-L103) can be used to solve this issue.

(BTW I think it's a bad idea to leave `p.Handle == nil` for skip-grant-table）

--- a/privilege/privileges/privileges.go
+++ b/privilege/privileges/privileges.go
@@ -222,6 +222,10 @@ func (p *UserPrivileges) GetEncodedPassword(user, host string) string {
 
 // GetAuthPlugin gets the authentication plugin for the account identified by the user and host
 func (p *UserPrivileges) GetAuthPlugin(user, host string) (string, error) {
+	if SkipWithGrant {
+		return mysql.AuthNativePassword, nil
+	}
+
 	mysqlPriv := p.Handle.Get()
 	record := mysqlPriv.connectionVerification(user, host)
 	if record == nil {

